File: agents/profiler/src/consumer.py
"""Profiler SQS consumer."""

from __future__ import annotations

import sys
import pathlib
from typing import Any

sys.path.insert(
    0,
    str(pathlib.Path(__file__).resolve().parents[2] / "shared" / "python" / "src"),
)
from sqs_consumer import run_consumer  # noqa: E402

from . import processor  # noqa: E402


def start_consumer(
    *,
    queue_url: str,
    pool: Any,
    logger: Any,
    concurrency: int = 2,
) -> Any:
    async def handler(message: dict[str, Any]) -> None:
        await processor.process_job(message, pool=pool, logger=logger)

    return run_consumer(
        queue_url=queue_url,
        handler=handler,
        concurrency=concurrency,
        logger=logger,
    )

File: agents/writer/src/consumer.js
// Writer SQS consumer.
//
// Thin wrapper over the shared runConsumer. Binds the processor to the
// writer's queue URL and injects the per-service deps.

const { runConsumer } = require("../../shared/node/src/sqs");
const { processJob } = require("./processor");

/**
 * @param {{
 *   queueUrl: string,
 *   anthropic: import('@anthropic-ai/sdk').default,
 *   pool: import('pg').Pool,
 *   logger: { info: Function, error: Function, warn: Function, debug: Function },
 *   concurrency?: number,
 * }} opts
 */
async function startWriterConsumer(opts) {
  const { queueUrl, anthropic, pool, logger, concurrency = 2 } = opts;

  return runConsumer({
    queueUrl,
    concurrency,
    logger,
    async handler(message) {
      await processJob(message, { anthropic, pool, logger });
    },
  });
}

module.exports = { startWriterConsumer };
